src/libraries/Microsoft.Extensions.Caching.Abstractions/src/DistributedCacheEntryOptions.cs

```diff
@@ -10,34 +10,29 @@ namespace Microsoft.Extensions.Caching.Distributed
     /// </summary>
     public class DistributedCacheEntryOptions
     {
-        private DateTimeOffset? _absoluteExpiration;
-        private TimeSpan? _absoluteExpirationRelativeToNow;
-        private TimeSpan? _slidingExpiration;
+        private DateTimeOffset _absoluteExpiration;
+        private bool _absoluteExpirationSet;
```

Comment:
If you use multiple bools, you should align them sequentially, because of alignment/padding.

The code as written: 56 bytes payload. Moving all the `bool` fields next to eachother: 40 payload bytes.

Reply:
Huh, fair enough. TIL. I guess normally when I'm looking that hard I'm dealing with a `struct` :)

Reply:
How do you find out the size of an object at runtime?

Reply:
I tried with VS debug and took a snapshot, also tried with GC.GetTotalMemory.  It tells me the size of my custom type is 56 with both ordering of fields. 

```cs
static void Main(string[] args)
{
    long memoryBefore = GC.GetTotalMemory(true);
    DistributedCacheEntryOptions myObject = new DistributedCacheEntryOptions();
    long memoryAfter = GC.GetTotalMemory(true);
    long objectSize = memoryAfter - memoryBefore;
    Console.WriteLine($"Memory size of MyClass: {objectSize} bytes");
}
```